tcf_core/context_processors.py

```diff
@@ -10,3 +12,36 @@ def base(request):
         'DEBUG': settings.DEBUG,
         'USER': request.user,
         'LATEST_SEMESTER': Semester.latest()}
+
+
+def history_cookies(request):
+    """Puts history from cookies into context variables."""
+    if 'previous_paths' in request.COOKIES:
+        previous_paths = request.COOKIES['previous_paths']
+        previous_paths = ast.literal_eval(previous_paths)
+    else:
+        previous_paths = ''
+
+    if 'previous_paths_titles' in request.COOKIES:
+        previous_paths_titles = request.COOKIES['previous_paths_titles']
+        previous_paths_titles = ast.literal_eval(previous_paths_titles)
+    else:
+        previous_paths_titles = ''
+
+    if 'count' in request.COOKIES:
+        count = request.COOKIES['count']
+    else:
+        count = 0
+
+    previous_paths_titles = [title[:80] + '...' for title in previous_paths_titles]
+
+    previous_paths_and_titles = None
+    if len(previous_paths) > 0 and len(previous_paths_titles) > 0:
+        previous_paths_and_titles = zip(previous_paths, previous_paths_titles)
+
+    return {
+        'previous_paths': previous_paths,
+        'count': count,
```

Comment:
I don't see where this `count` variable is being used, is it necessary? Is it always the same as `len(previous_paths)`? Also is `previous_paths` and `previous_path_titles` always going to be the same length?